### DISCUSSION TITLE
Feedback on the new Graph / Custom Workflow feature

### DISCUSSION THREAD

**Discussion by patrickchamp** [Acly/krita-ai-diffusion#1374] [Ideas]
I absolutely love this feature btw. I finalized a workflow and thought i'd send you feedback. I think just updating the tooltips could resolve a few of these.

- When i click "Open WebUI to create custom workflows", I expected to be able to edit the currently selected workflow. e.g. I expected to click that button, make a small edit to the workflow, then click save. it would be a nice quality of life improvement, but just a tooltip with more info would make the learning curve smoother.
- Similar issue: When i click "Open WebUI to create custom workflows", I expected to be able to click "Generate" in krita and that it would execute the workflow in the tab it had just been opened. Instead, I had to select a graph and click through 5-10 "Comfy UI Web (3k8dh59d...)" options, then click generate after each one to figure out if the one i had selected was linked to the ComfyUI tab it had just opened. Because i was learning, I clicked the "Open WebUI" button several times and it kept adding new items to the selection list, which made it hard to ramp up and grasp the functionality. If I close all my comfyui tabs, the list doesn't get cleared out, so it gets more confusing as you're getting the hang of it. What made it more confusing was that some of my workflows didn't work, so clicking generate and seeing what came out didn't really help me figure out whether i had selected the workflow that matched the comfyui tab i had open. At a minimum, it would help that if you click "Open WebUI" it would auto-select the correct "ComfyUI Web" dropdown.
- The workflow i created had about 16 parameters. The sliders made it difficult to reset a value to 0. The closest i could get was values like 0.27, then i'd have to click to the left of the slider to get it down to 0.7, then my only option was to use the keyboard keys to adjust it by 0.01 increments and finally get it to 0. This was fine once i figured it out, but obviously not ideal for less resourceful users.
- For my use case, I would have found a "reset parameters to default" button helpful since i had so many parameters.

What i loved:

- The graph feature in general. I'm not sure how many people needed an expression editor, so the fact that i could add it myself was fun and awesome.
- The parameter nodes were unexpectedly delightful to add. i like that they auto named themselves, auto detected the min/max values, and were instantly visible in Krita. 
- I love that you don't need to edit the json to make the node index numbers match what the plugin needs. You just add the output node and one or two other nodes and it all just works. Beautiful stuff.
- Great job on this.

**Comment by patrickchamp** [comment]
One more thought. It would be great to know where you can find workflows others have created and shared. I just discovered the discord which is great. Ideally, there would be a little "Discover workflows" button that would send you to something like a civitai search result page that queries for workflows tagged with krita. It would be nice to recommend/direct people to one location where we can share/discover/discuss workflows. Just a thought.

**Comment by Acly** [comment]
Those are some good points, and some I've thought about when implementing it. The interactions with the web UI are going to be tricky though, because to open specific workflows it would have to support that in the first place. Currently there is just no mechanism to do it as far as I know.

* _Open current workflow in web UI:_ unlikely for now
* _Select ComfyUI tab after openinig web UI:_ this should sort of work in [v1.28.0](https://github.com/Acly/krita-ai-diffusion/releases/tag/v1.28.0), it's fuzzy because it takes a while for the browser tab to open, and you also need the Krita output node before any workflow sharing happens
* _Clear tabs after closing:_ unlikely for now. Don't have that information, ComfyUI doesn't expose that.
* _Parameter management_: [v1.28.0](https://github.com/Acly/krita-ai-diffusion/releases/tag/v1.28.0) has groups which can be reset. Didn't add a global reset because didn't know where to put the button :)
